pkg/cvo/metrics.go

```diff
@@ -0,0 +1,52 @@
+package cvo
+
+import (
+	"github.com/openshift/cluster-version-operator/lib/resourcemerge"
+	osv1 "github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func (optr *Operator) registerMetrics() error {
+	m := newOperatorMetrics(optr)
+	return prometheus.Register(m)
+}
+
+type operatorMetrics struct {
+	optr *Operator
+
+	version *prometheus.GaugeVec
+}
+
+func newOperatorMetrics(optr *Operator) *operatorMetrics {
+	return &operatorMetrics{
+		optr: optr,
+
+		version: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "cluster_version",
+			Help: "Reports the version of the cluster.",
+		}, []string{"type", "version", "payload"}),
+	}
+}
+
+func (m *operatorMetrics) Describe(ch chan<- *prometheus.Desc) {
+	ch <- m.version.WithLabelValues("", "", "").Desc()
+}
+
+func (m *operatorMetrics) Collect(ch chan<- prometheus.Metric) {
+	current := m.optr.currentVersion()
+	g := m.version.WithLabelValues("current", current.Version, current.Payload)
+	g.Set(1)
+	ch <- g
+	if cv, err := m.optr.cvoConfigLister.Get(m.optr.name); err == nil {
+		if update := cv.Spec.DesiredUpdate; update != nil && update.Payload != current.Payload {
+			g = m.version.WithLabelValues("update", update.Version, update.Payload)
```

Comment:
it is usual to just reuse the same variable when doing this metric push?

Reply:
no, I'll fix that